src/navigation/context.py

```diff
@@ -205,15 +205,28 @@ def convert(waypoint: GPSWaypoint) -> Waypoint:
     return Waypoint(fiducial_id=waypoint.id, tf_id=f"course{waypoint.id}", type=waypoint.type), SE3(position=odom)
 
 
+def convert_cartesian_to_gps(coordinate: np.ndarray) -> GPSWaypoint:
+    """
+    Converts a coordinate to a GPSWaypoint (used for sending data back to basestation)
+    """
+    lat, lon, _ = pymap3d.enu2geodetic(
+        e=coordinate[0], n=coordinate[1], u=coordinate[2], lat0=REF_LAT, lon0=REF_LON, h0=0.0, deg=True
```

Comment:
I think so, it works fine when we do it like that for other things (conversions in debug_course_publisher)

Reply:
nah it hasn't been tested with teleop GUI I don't think its quite ready yet I tested by echoing the topic. I feel like its probably fine to just merge and test with the GUI on the rover but if we feel like we need to merge in the GUI when its done and test first that's fine too

Reply:
Wait then how did you verify that the outputted gps coords are correct?

Reply:
I didn't, we're using the same conversion that we use for our debug_course_publisher (which converts odom -> GPS -> odm again) which works fine so I'm just assuming that it works

Reply:
hmm ok, speaking of that, I noticed some differences between this function and that function:
```py
def convert_waypoint_to_gps(waypoint_pose_pair: Tuple[Waypoint, SE3]) -> GPSWaypoint:
    waypoint, pose = waypoint_pose_pair
    lat, lon, _ = pymap3d.enu2geodetic(pose.position[0], pose.position[1], pose.position[1], REF_LAT, REF_LON, 0.0)
    return GPSWaypoint(lat, lon, waypoint.type, waypoint.fiducial_id)
```
why is this one using pose.position[1] for both y and z, and why is deg only set to true in one of them?

Reply:
lol bit of an oops moment. I suspect that we never noticed any issues because in the debug course publisher the value of those points is too small to cause any difference

Reply:
as in we're always doing stuff by the origin. I'll fix it though nice catch

Reply:
deg is default to true so it doesn't really matter

Reply:
I mean we can wait for the GUI change but I'm not sure that we're really going to be able to verify that the coordinates look correct unless we run it on the rover regardless since in the sim its going to be pretty hard to see that they're in the right spot relative to the reference coordinate and gate in gazebo